transactions/transactor.go

```diff
@@ -334,7 +334,7 @@ func (t *Transactor) HashTransaction(args SendTxArgs) (validatedArgs SendTxArgs,
 	newNonce := hexutil.Uint64(nonce)
 	newGas := hexutil.Uint64(gas)
 	validatedArgs.Nonce = &newNonce
-	if args.GasPrice != nil {
+	if args.IsDynamicFeeTx() {
 		validatedArgs.GasPrice = (*hexutil.Big)(gasPrice)
 	} else {
 		validatedArgs.MaxPriorityFeePerGas = (*hexutil.Big)(gasTipCap)
```

Comment:
Overwriting `validatedArgs.MaxPriorityFeePerGas` twice seems wrong. I think the author wanted to sum them up as expected with `MaxPriorityFeePerGas`

Reply:
yes, you're right! good catch!